capstone/src/test/java/com/google/sps/ClubServletTest.java

```diff
@@ -0,0 +1,132 @@
+package com.google.sps.servlets;
+
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.tools.development.testing.LocalBlobstoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+@RunWith(JUnit4.class)
+public class ClubServletTest {
+
+  private final String SAMPLE_CLUB_NAME = "Club 1";
+  private final String SAMPLE_CLUB_DESC_1 = "Test club description";
+  private final String SAMPLE_CLUB_DESC_2 = "Another club description";
+  private final String SAMPLE_CLUB_WEB = "www.test-club.com";
+  private final BlobKey SAMPLE_BLOB = new BlobKey("test-blobkey");
+  private final String TEST_EMAIL = "test-email@gmail.com";
+  private final ImmutableList<String> STUDENT_LIST = ImmutableList.of(TEST_EMAIL);
+  private final String VALID_URL = "/registration-msg.html?is-valid=true";
+  private final String INVALID_URL = "/registration-msg.html?is-valid=false";
```

Comment:
Quick question: are these constants specific to the individual test use cases? If so, it's actually better to define them or even hard-code them in the specific test. Test constants are generally not used except for commonly shared cases across multiple tests (ex: the email of the user is always going to return the same thing if we're using the same user in all the tests).

If it's for set-up though, it's ok to leave it as-is.